docs/user-guide/data-structures/creating-arrays-and-datasets.ipynb

```diff
@@ -0,0 +1,527 @@
+    "Alternatively, passing a NumPy array:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "a = np.array([[1, 2], [3, 4]])\n",
+    "v = sc.array(dims=['x', 'y'], values=a)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that *both* the NumPy array and Python lists are copied into the Scipp variable which leads to some additional memory usage.\n",
+    "\n",
+    "The `dtype` and unit of the variable are deduced automatically in the above cases:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "v"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If required, they can be specified directly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.array(dims=['x', 'y'], values=[[1, 2], [3, 4]], dtype='float64', unit='m')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Variances can be added using the `variances` keyword:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "sc.array(dims=['x', 'y'], values=[[1.0, 2.0], [3.0, 4.0]],\n",
+    "         variances=[[0.1, 0.2], [0.3, 0.4]])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Scalars: 0-D Variables\n",
+    "\n",
+    "Scalars are variables with no dimensions.\n",
+    "They can be constructed using, among other functions, [scipp.scalar](../../generated/functions/scipp.scalar.rst):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.scalar(3.41)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`scipp.scalar` will always produce a scalar variable, even when passed a sequence like a list:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.scalar([3.41])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this case, it stores the Python list as-is in a Scipp variable.\n",
+    "\n",
+    "Multiplying or dividing a value by a unit also produces a scalar variable:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "4.2 * sc.units.m"
```

Comment:
In the past year or so I moved way from using the constants in `sc.units`. I felt that it requires extra memorization by the user (which constants exist?). Instead I am usually favoring `sc.Unit(string)`. What do you think?

Reply:
I essentially never use this multiplication operator and only specify units as strings in `unit=` args. Or use `sc.Unit(string)` for comparing units.